libcodechecker/analyze/analysis_manager.py

```diff
@@ -433,6 +464,26 @@ def check(check_data):
         return 1, skipped, reanalyzed, action.analyzer_type, None
 
 
+def create_actions_map(actions):
+    """
+    Create a dict for the build actions.
+    Key of the dict: (source_file, target).
+    """
+    result = dict()
+    for act in actions:
+        if act.source_count > 1:
```

Comment:
I think this filtering and splitting up is already done in the logger. If there are multiple source files in one build action it will be added as multiple build actions into the compile command json file. Do we have to check this again here?

Reply:
It is just some sanity check for asserting invariants and alerting on possible programming errors.
First I wanted `assert` or `sys.exit` but later I decided we just log and survive by choosing the first source.